src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -470,7 +475,7 @@ public function setStatusCode(int $code, $text = null): object
         }
 
         if (null === $text) {
-            $this->statusText = isset(self::$statusTexts[$code]) ? self::$statusTexts[$code] : 'unknown status';
+            $this->statusText = isset(self::STATUS_TEXTS[$code]) ? self::STATUS_TEXTS[$code] : 'unknown status';
```

Comment:
It is not planned anywhere I guess but the public static could have been overwritten.
To really not break anything and to make this change even more unlikely to happen (😎) we would need to use the deprecated property to read from.

Reply:
The status texts could have been extended, for instance if an application uses non-standard response codes.

That static array might not look like an intended extension point. However an application might use it that way. With that in mind, the change you're proposing could be considered a BC break.

Reply:
if the value differs between the const and the property we should trigger a deprecation, everywhere this property is used.

Reply:
> though i agree Response::$statusTexts could be a mutable extension point. Not sure what to do with it actually :)

To take care of this we need to leave the deprecated property in those places. Users can still migrate to the new const for the next major version when the old property will be removed.

> if the value differs between the const and the property we should trigger a deprecation, everywhere this property is used.

Triggering deprecation messages if reading from normally unknown array indexes is found is an interesting idea. I just don't know if this is really necessary. Already "regular" read-only use of the property have to migrate to the const anyway.

Reply:
> . Users can still migrate to the new const for the next major version when the old property will be removed.

i meant, it can be a legit feature per case maybe - something we want to keep (not that im a fan of such global state features)

however, in this case, the `$text` argument in `setStatusCode` is probably the better extension point :+1:

Reply:
Yeah right. We should indeed make that clear with a comment or setter method then.